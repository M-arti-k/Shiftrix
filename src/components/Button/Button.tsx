import type { FC, JSX } from "react";
import type { IButton } from "src/interfaces/IButton";
import styles from "./Button.module.scss";
import clsx from "clsx";

const Button: FC<IButton> = ({
  children,
  type,
  variant,
  color,
}): JSX.Element => {
  return (
    <button
      data-testid="Button"
      className={clsx(
        styles["button-base"],
        styles[`${variant}`],
        styles[`${color}`]
      )}
      type={type}
    >
      {children}
    </button>
  );
};

export default Button;
