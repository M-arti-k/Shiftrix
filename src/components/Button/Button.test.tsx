import { render, screen } from "@testing-library/react";
import Button from "./Button";
import "@testing-library/jest-dom";

describe("Button Testing", (): void => {
  let btn: HTMLElement;

  describe("Default Button (type=button, variant=long)", (): void => {
    beforeEach((): void => {
      render(
        <Button type="button" variant="long">
          Test
        </Button>
      );
      btn = screen.getByText("Test");
    });

    test("Button renders", () => {
      expect(btn).toBeInTheDocument();
    });

    test("Button contains correct text", (): void => {
      expect(btn).toHaveTextContent("Test");
    });

    test("Button matches snapshot", (): void => {
      expect(btn).toMatchSnapshot();
    });
  });

  describe("Button with type=submit", (): void => {
    beforeEach((): void => {
      render(
        <Button variant="long" type="submit" data-testid="Button">
          Test
        </Button>
      );
      btn = screen.getByTestId("Button");
    });

    test("Button receives type attribute", (): void => {
      expect(btn).toHaveAttribute("type", "submit");
    });
  });

  describe("Button with variant=long", (): void => {
    beforeEach((): void => {
      render(
        <Button variant="long" type="button" data-testid="Button">
          Test
        </Button>
      );
      btn = screen.getByTestId("Button");
    });

    test("Button receives variant class", (): void => {
      expect(btn).toHaveClass("long");
    });
  });
});
