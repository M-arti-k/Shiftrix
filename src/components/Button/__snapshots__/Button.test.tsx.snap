// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Testing Default Button (type=button, variant=long) Button matches snapshot 1`] = `
<button
  class="button-base long undefined"
  data-testid="Button"
  type="button"
>
  Test
</button>
`;
